hyperspy/_signals/complex_signal.py

```diff
@@ -0,0 +1,252 @@
+            self.data = self.data.astype(dtype)
+        else:
+            raise AttributeError('Complex data can only be converted into other complex dtypes!')
+
+    def angle(self, deg=False):
+        """Return the angle (also known as phase or argument). If the data is real, the angle is 0
+        for positive values and 2$\pi$ for negative values.
+
+        Parameters
+        ----------
+        deg : bool, optional
+            Return angle in degrees if True, radians if False (default).
+
+        Returns
+        -------
+        angle : HyperSpy signal
+            The counterclockwise angle from the positive real axis on the complex plane,
+            with dtype as numpy.float64.
+
+        """
+        sig = self._deepcopy_with_new_data(np.angle(self.data, deg))
+        sig.set_signal_type('')
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'angle({})'.format(title)
+        return sig
+
+    def unwrapped_phase(self, wrap_around=False, seed=None, show_progressbar=None):
+        """Return the unwrapped phase as an appropriate HyperSpy signal.
+
+        Parameters
+        ----------
+        wrap_around : bool or sequence of bool, optional
+            When an element of the sequence is  `True`, the unwrapping process
+            will regard the edges along the corresponding axis of the image to be
+            connected and use this connectivity to guide the phase unwrapping
+            process. If only a single boolean is given, it will apply to all axes.
+            Wrap around is not supported for 1D arrays.
+        seed : int, optional
+            Unwrapping 2D or 3D images uses random initialization. This sets the
+            seed of the PRNG to achieve deterministic behavior.
+        show_progressbar : None or bool
+            If True, display a progress bar. If None the default is set in
+            `preferences`.
+
+        Returns
+        -------
+        phase_image: :class:`~hyperspy._signals.Signal2D`
+            Unwrapped phase.
+
+        Notes
+        -----
+        Uses the :func:`~skimage.restoration.unwrap_phase` function from `skimage`.
+        The algorithm is based on Miguel Arevallilo Herraez, David R. Burton, Michael J. Lalor,
+        and Munther A. Gdeisat, “Fast two-dimensional phase-unwrapping algorithm based on sorting
+        by reliability following a noncontinuous path”, Journal Applied Optics,
+        Vol. 41, No. 35, pp. 7437, 2002
+
+        """
+        from skimage.restoration import unwrap_phase
+        phase = self.phase
+        phase.map(unwrap_phase, wrap_around=wrap_around, seed=seed,
+                  show_progressbar=show_progressbar)
+        return phase  # Now unwrapped!
+
+    def plot(self, navigator="auto", axes_manager=None, representation='cartesian', **kwargs):
+        """Plot the signal at the current coordinates.

```

Comment:
Instead of copy/paste I think we should move this docstring to [docstring](https://github.com/hyperspy/hyperspy/tree/master/hyperspy/docstrings). [Example of usage](https://github.com/hyperspy/hyperspy/blob/67f96bea30b344cc2296408f4725774e2e24f6f8/hyperspy/signal.py#L2178)